docs/guides/deployment/overriding-job-variables.md

```diff
@@ -0,0 +1,177 @@
+
+Job variables are just infrastructure related values that are configurable on a work pool, which may be relevant to how your flow run executes on your infrastructure.
+
+<hr>
+
+Let's use `env` - the only job variable that is configurable for all work pool types - as an example.
+
+When you create or edit a work pool, you can specify a set of environment variables that will be set in the runtime environment of the flow run.
+
+For example, you might want a certain deployment to have the following environment variables available:
+```json
+{
+  "EXECUTION_ENV": "staging",
+  "MY_NOT_SO_SECRET_CONFIG": "some-value",
+}
+```
+
+So, as opposed to hardcoding these values into your work pool in the UI and making them available to all deployments associated with that work pool, you can override these values on a _per-deployment basis_.
+
+Let's look at how to do that.
+
+## How to override job variables
+Say we have the following repo structure:
+```
+» tree
+.
+├── README.md
+├── requirements.txt
+├── demo_project
+│   ├── daily_flow.py
+```
+
+... and we have some `demo_flow.py` file like this:
+
+```python
+import os
+from prefect import flow, task
+
+@task
+def do_something_important(not_so_secret_value: str) -> None:
+    print(f"Doing something important with {not_so_secret_value}!")
+
+@flow(log_prints=True)
+def some_work():
+    environment = os.environ.get("EXECUTION_ENVIRONMENT", "local")
+    
+    print(f"Coming to you live from {environment}!")
+    
+    not_so_secret_value = os.environ.get("MY_NOT_SO_SECRET_CONFIG")
+    
+    if not_so_secret_value is None:
+        raise ValueError("You forgot to set MY_NOT_SO_SECRET_CONFIG!")
+
+    do_something_important(not_so_secret_value)
+```
+### Using a `prefect.yaml` file
+In this case, let's also say we have the following deployment definition in a `prefect.yaml` file at the root of our repository:
+```yaml
+deployments:
+- name: demo-deployment
+  entrypoint: demo_project/demo_flow.py:some_work
+  work_pool:
+    name: local
+  schedule: null
+```
+
+!!! note
+    While not the focus of this guide, note that this deployment definition uses a default "global" `pull` step, because one is not explicitly defined. For reference, here's what that would look like at the top of the `prefect.yaml` file:
+    ```yaml
+    pull:
+    - prefect.deployments.steps.git_clone: &clone_repo
+        repository: https://github.com/some-user/prefect-monorepo
+        branch: main
+    ```
+
+To provide the `EXECUTION_ENVIRONMENT` and `MY_NOT_SO_SECRET_CONFIG` environment variables to this deployment, we can add a `job_variables` section to our deployment definition in the `prefect.yaml` file:
+
+#### Hard-coded job variables
+```yaml
+deployments:
+- name: demo-deployment
+  entrypoint: demo_project/demo_flow.py:some_work
+  work_pool:
+    name: local
+    job_variables:
+        env:
+            EXECUTION_ENVIRONMENT: staging
+            MY_NOT_SO_SECRET_CONFIG: plumbus
```

Comment:
```suggestion
            MY_NOT_SO_SECRET_CONFIG: some-value
```

Reply:
![image](https://github.com/PrefectHQ/prefect/assets/31014960/7f104467-3575-444e-a94c-4a25b0c3e622)

Reply:
lol then make it plumbus everywhere. I don't care what the value is but make it the same throughout the doc

Reply:
[81451f9](https://github.com/PrefectHQ/prefect/pull/12033/commits/81451f99e289a6a917c97c09d041f94cc7d49d44)